test/unit/test_borealis.py

```diff
@@ -0,0 +1,398 @@
+                          '../test_files/somefile.rawacf')
+
+    def test_empty_file(self):
+        """
+        Tests if BorealisRead constructor with an empty file
+
+        Expected behaviour: raise OSError, unable to open (file signature not found)
+        """
+        self.assertRaises(OSError,
+                          pydarn.BorealisRead, '../test_files/empty.rawacf')
+
+    def test_open_rawacf(self):
+        """
+        Tests BorealisRead constructor on opening a rawacf.
+        It should be able to open the file, read it and convert to dictionary
+
+        Checks:
+            - group names is a list, greater than length 0
+        """
+        file_path = borealis_rawacf_file
+        reader = pydarn.BorealisRead(file_path)
+        self.assertIsInstance(reader.group_names, list)
+        self.assertGreater(len(reader.group_names), 0)
+
+    def test_open_bfiq(self):
+        """
+        Tests BorealisRead constructor on opening a iqdat.
+        It should be able to open the file, read it and convert to dictionary
+
+        Checks:
+            - group names is a list, greater than length 0
+        """
+        file_path = borealis_bfiq_file
+        reader = pydarn.BorealisRead(file_path)
+        self.assertIsInstance(reader.records, dict)
+        self.assertGreater(len(reader.group_names), 0)
+
+    # TODO: potential issue if files change and the values are not the same :/
+    def test_read_bfiq(self):
+        """
+        Test reading records from bfiq.
+
+        Checks:
+            - returns correct data structures
+            - returns expected values
+        """
+        file_path = borealis_bfiq_file
+        dm = pydarn.BorealisRead(file_path)
+        records = dm.read_bfiq()
+        first_record = records[dm.group_names[0]]
+        self.assertIsInstance(records, collections.OrderedDict)
+        self.assertIsInstance(first_record, dict)
+        self.assertIsInstance(first_record['num_slices'], np.int64)
+
+    def test_read_rawacf(self):
+        """
+        Test reading records from rawacf.
+
+        Checks:
+            - returns correct data structures
+            - returns expected values
+        """
+        file_path = borealis_rawacf_file
+        dm = pydarn.BorealisRead(file_path)
+        records = dm.read_rawacf()
+        first_record = records[dm.group_names[0]]
+        self.assertIsInstance(records, collections.OrderedDict)
+        self.assertIsInstance(first_record, dict)
+        self.assertIsInstance(first_record['num_slices'], np.int64)
+
+
+class TestBorealisWrite(unittest.TestCase):
+    """
+    Tests BorealisWrite class
+    """
+
+    def setUp(self):
+        pass
+
+    def test_borealis_write_constructor(self):
+        """
+        Tests BorealisWrite constructor
+
+        Expected behaviour
+        ------------------
+        Contains file name of the data if given to it.
+        """
+        rawacf_data = copy.deepcopy(
```

Comment:
You could put this line in setUp, which is something I am going to do for DMap IO and SuperDARN IO

Reply:
done